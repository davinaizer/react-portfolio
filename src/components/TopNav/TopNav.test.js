import React from 'react';
import TopNav from './TopNav';

import { shallow } from 'enzyme';

it('should render correctly with no props', () => {
  const component = shallow(<TopNav />);
  expect(component).toMatchSnapshot();
});

it('should render with given routes object', () => {
  const routes = [
    { path: '/', title: 'Home', isVisible: false },
    { path: '/about', title: 'About' },
  ];
  const component = shallow(<TopNav links={routes} />);
  // expect(component.contains(<Button />)).toEqual(true);
  expect(component).toMatchSnapshot();
});
