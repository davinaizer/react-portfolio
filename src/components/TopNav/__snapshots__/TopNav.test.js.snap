// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "<div />",
  Symbol(enzyme.__nodes__): Array [
    "<div />",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should render with given routes object 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopNav
    links={
      Array [
        Object {
          "isVisible": false,
          "path": "/",
          "title": "Home",
        },
        Object {
          "path": "/about",
          "title": "About",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Navbar
        className="app-navbar"
        expand="md"
        fixed="top"
        tag="nav"
      >
        <NavLink
          activeClassName=""
          className="navbar-brand"
          tag="a"
          to="/"
        >
          <p>
            DN
          </p>
        </NavLink>
        <NavbarToggler
          onClick={[Function]}
          tag="button"
          type="button"
        />
        <Collapse
          appear={false}
          enter={true}
          exit={true}
          in={false}
          isOpen={false}
          mountOnEnter={false}
          navbar={true}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          tag="div"
          timeout={350}
          unmountOnExit={false}
        >
          <Nav
            className="ml-auto"
            navbar={true}
            onClick={[Function]}
            tag="ul"
            vertical={false}
          >
            <NavItem
              tag="li"
            >
              <NavLink
                activeClassName="isActive"
                className="nav-link"
                tag="a"
                to="/about"
              >
                About
              </NavLink>
            </NavItem>
          </Nav>
        </Collapse>
      </Navbar>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName=""
            className="navbar-brand"
            tag="a"
            to="/"
          >
            <p>
              DN
            </p>
          </NavLink>,
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
          />,
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              className="ml-auto"
              navbar={true}
              onClick={[Function]}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="isActive"
                  className="nav-link"
                  tag="a"
                  to="/about"
                >
                  About
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>,
        ],
        "className": "app-navbar",
        "expand": "md",
        "fixed": "top",
        "tag": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeClassName": "",
            "children": <p>
              DN
            </p>,
            "className": "navbar-brand",
            "tag": "a",
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "DN",
            },
            "ref": null,
            "rendered": "DN",
            "type": "p",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "tag": "button",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appear": false,
            "children": <Nav
              className="ml-auto"
              navbar={true}
              onClick={[Function]}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="isActive"
                  className="nav-link"
                  tag="a"
                  to="/about"
                >
                  About
                </NavLink>
              </NavItem>
            </Nav>,
            "enter": true,
            "exit": true,
            "in": false,
            "isOpen": false,
            "mountOnEnter": false,
            "navbar": true,
            "onEnter": [Function],
            "onEntered": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "onExiting": [Function],
            "tag": "div",
            "timeout": 350,
            "unmountOnExit": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="isActive"
                    className="nav-link"
                    tag="a"
                    to="/about"
                  >
                    About
                  </NavLink>
                </NavItem>,
              ],
              "className": "ml-auto",
              "navbar": true,
              "onClick": [Function],
              "tag": "ul",
              "vertical": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "/about",
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    activeClassName="isActive"
                    className="nav-link"
                    tag="a"
                    to="/about"
                  >
                    About
                  </NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "activeClassName": "isActive",
                    "children": "About",
                    "className": "nav-link",
                    "tag": "a",
                    "to": "/about",
                  },
                  "ref": null,
                  "rendered": "About",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Navbar
          className="app-navbar"
          expand="md"
          fixed="top"
          tag="nav"
        >
          <NavLink
            activeClassName=""
            className="navbar-brand"
            tag="a"
            to="/"
          >
            <p>
              DN
            </p>
          </NavLink>
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
          />
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              className="ml-auto"
              navbar={true}
              onClick={[Function]}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="isActive"
                  className="nav-link"
                  tag="a"
                  to="/about"
                >
                  About
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName=""
              className="navbar-brand"
              tag="a"
              to="/"
            >
              <p>
                DN
              </p>
            </NavLink>,
            <NavbarToggler
              onClick={[Function]}
              tag="button"
              type="button"
            />,
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Nav
                className="ml-auto"
                navbar={true}
                onClick={[Function]}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="isActive"
                    className="nav-link"
                    tag="a"
                    to="/about"
                  >
                    About
                  </NavLink>
                </NavItem>
              </Nav>
            </Collapse>,
          ],
          "className": "app-navbar",
          "expand": "md",
          "fixed": "top",
          "tag": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeClassName": "",
              "children": <p>
                DN
              </p>,
              "className": "navbar-brand",
              "tag": "a",
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "DN",
              },
              "ref": null,
              "rendered": "DN",
              "type": "p",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "tag": "button",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "appear": false,
              "children": <Nav
                className="ml-auto"
                navbar={true}
                onClick={[Function]}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="isActive"
                    className="nav-link"
                    tag="a"
                    to="/about"
                  >
                    About
                  </NavLink>
                </NavItem>
              </Nav>,
              "enter": true,
              "exit": true,
              "in": false,
              "isOpen": false,
              "mountOnEnter": false,
              "navbar": true,
              "onEnter": [Function],
              "onEntered": [Function],
              "onEntering": [Function],
              "onExit": [Function],
              "onExited": [Function],
              "onExiting": [Function],
              "tag": "div",
              "timeout": 350,
              "unmountOnExit": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      activeClassName="isActive"
                      className="nav-link"
                      tag="a"
                      to="/about"
                    >
                      About
                    </NavLink>
                  </NavItem>,
                ],
                "className": "ml-auto",
                "navbar": true,
                "onClick": [Function],
                "tag": "ul",
                "vertical": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "/about",
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavLink
                      activeClassName="isActive"
                      className="nav-link"
                      tag="a"
                      to="/about"
                    >
                      About
                    </NavLink>,
                    "tag": "li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "activeClassName": "isActive",
                      "children": "About",
                      "className": "nav-link",
                      "tag": "a",
                      "to": "/about",
                    },
                    "ref": null,
                    "rendered": "About",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
